src/features/auth/components/Guard.tsx

```diff
@@ -0,0 +1,18 @@
+import { Outlet } from 'react-router-dom';
+import { Navigate } from 'react-router-dom';
+
+import { useGetUser } from '../api/getUser';
+
+const Guard = () => {
+  const { data, isPending, isSuccess } = useGetUser();
+
+  console.log(data);
+
+  if (isPending) {
+    return <div>Loading...</div>;
+  }
+
+  return isSuccess && data ? <Outlet /> : <Navigate to="/signin" />;
```

Comment:
Jest spoko. Na tym etapie jeszcze mozna pomyslec o wyswietleniu strony, lub przekierowaniu na strone braku uprawnieni (jak posprawdzasz czy user ma odpowiednia role do strony w jego danych) - ale to juz optional bo nie wiem czy bedziesz mial role:)